test/functional/test_framework/util.py

```diff
@@ -385,6 +385,17 @@ def set_node_times(nodes, t):
     for node in nodes:
         node.setmocktime(t)
 
+def disconnect_nodes(from_connection, node_num):
+    timeout = 5
+    while True:
+        peer_ids = [peer['id'] for peer in from_connection.getpeerinfo() if "testnode%d" % node_num in peer['subver']]
+        if not peer_ids:
+            break
+        [from_connection.disconnectnode(nodeid=peer_id) for peer_id in peer_ids]
+        assert timeout > 0.1, "timed out waiting for disconnect"
```

Comment:
This seems like it would be made better as the for loop condition:

    for i in range(50): 
        ...
        time.sleep(0.1)
    else:
        assert 0, "timed out waiting for disconnect"

At least for me, that's a bit easier to read.